packages/core-data/src/types.ts

```diff
@@ -0,0 +1,85 @@
+export type Context = 'edit' | 'embed' | 'view';
+
+interface Edit {}
+
+// @ts-ignore
+export interface EntityRecord< C extends Context > {}
+
+export interface EntityConfig< Record extends EntityRecord<any> > {
+	/** Path in WP REST API from which to request records of this entity. */
+	baseURL: string;
+
+	/** Arguments to supply by default to API requests for records of this entity. */
+	baseURLParams: { context: Context };
+
+	/**
+	 * Returns the title for a given record of this entity.
+	 *
+	 * Some entities have an associated title, such as the name of a
+	 * particular template part ("full width") or of a menu ("main nav").
+	 */
+	getTitle( record: Record ): string;
+
+	/**
+	 * Indicates an alternate field in record that can be used for identification.
+	 *
+	 * e.g. a post has an id but may also be uniquely identified by its `slug`
+	 */
+	key: string;
+
+	/**
+	 * Collection in which to classify records of this entity.
```

Comment:
Another word would be a "namespace", since some frameworks refer to database tables as "collections". Namespace isn't perfect too, though – I guess we could use either.

Reply:
yeah it still puzzles me and I think the reason is that this `kind` value connotes different things at the same time. I'm reluctant to use the term namespace because I think that's a back-justification. one of the effects is namespacing, but it's really not used that way.

 - `namespace` _is_ [used as part of the API](https://github.com/WordPress/gutenberg/blob/0a41b4bb2edbfd1cdc390ed4d50f9c6eb784d806/packages/core-data/src/entities.js#L234) to get versions of the resources, e.g. `wp/v2` or `wpcom/v1.1`.
 - it's clearly being used as categorization or classification when contrasting `postType` and `taxonomy`. in cases such as in calling `getEntityRecord` it definitely feels more like "this general kind of entity record" as a grouping term rather than "the entity records provided by this plugin" which is generally how we use "namespace" elsewhere in the project

probably what's throwing us both off is that `getOrLoadEntitiesConfig` was built upon `kind` when in fact that function serves a subtly different purpose; more again an incidental design element related to optimization rather than intent.